backend/curriculum_tracking/models.py

```diff
@@ -772,7 +772,38 @@ def reviewer_user_names(self):
     @property
     def agile_card_status(self):
         return self.agile_card.status
+    
+    def request_user_can_add_review(self, user=None):
+        """
+        Check if current user can add a review to this project
+        This function is only used in template rendering and we are using threadlocals,
+        that is why we need to avoid argument parameters unless they're for testing purposes.
+        """
+        from threadlocal_middleware import get_current_user
+
+        card: AgileCard = self.agile_card
+
+        if card.status not in [
+            AgileCard.IN_REVIEW,
+            AgileCard.COMPLETE,
+            AgileCard.REVIEW_FEEDBACK,
+        ]:
+            return False
+
+        user = user or get_current_user()
 
+        if not user:
+            return False
+
+        if user.is_superuser:
+            return True
+
+        if user in card.reviewers.all():
+            return True
+
+        return not card.request_user_is_assignee(user) and any(
+            card.user_has_permission(user, perm) for perm in Team.PERMISSION_VIEW
```

Comment:
This doesn't look quite right. You are saying that anyone who is not the assignee and who is allowed to see the card is allowed to review the card. The code can also be a bit more clear.

Rather:

```
if card.request_user_is_assignee(user):
    return False

return any(card.user_has_permission(user, perm) in [PERMISSION_REVIEW_CARDS,PERMISSION_TRUSTED_REVIEWER])
```